frontend/src/components/templates/Form/HideSectionsContainer.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+  Children,
+  createContext,
+  FunctionComponent,
+  useEffect,
+  useState,
+} from 'react';
+import Button from '../../atoms/hyperTexts/Button';
+import Loader from '../../atoms/Loader';
+
+export type HideSectionsContextType = {
+  setReadyForNextSteps: (v: boolean) => void;
+};
+
+export const HideSectionsContext = createContext<HideSectionsContextType>({
+  setReadyForNextSteps: () => null,
+});
+
+type Props = {
+  SubmissionPanel: FunctionComponent;
+};
+
+export const HideSectionsContainer: FunctionComponent<Props> = ({
+  SubmissionPanel,
+  children,
+}) => {
+  const [showOnlyFirstStep, setShowOnlyFirstStep] = useState(false);
+  const [isShowNextStepButtonDisabled, setIsShowNextStepButtonDisabled] =
+    useState(false);
+  const [areNextStepsLoading, setAreNextStepLoading] = useState(false);
+  const [readyForNextSteps, setReadyForNextSteps] = useState(true);
+
+  useEffect(() => {
```

Comment:
This kind of useEffect is discouraged
See the Recap for useEffect doc here : https://beta.reactjs.org/learn/synchronizing-with-effects#recap
 - read the whole page if something is not clear
 - please do the challenges in the Recap section

The 2 useEffect in this component should be removed. The second one also has the same dependencies so will be triggered immediately following the first one. That could cause unexpected side effects and is hard to follow.